### PR TITLE
Record call results when removing an appointment from the overdue list

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.shortcut.com/simpledotorg/story/4669/record-a-callresult-whenever-a-call-is-made-in-contactpatientbottomsheet

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt [hunk 1]
```diff
@@ -56,7 +57,7 @@ class RemoveOverdueEffectHandler @AssistedInject constructor(
     return ObservableTransformer { effects ->
       effects
           .observeOn(schedulersProvider.io())
-          .doOnNext { appointmentRepository.cancelWithReason(it.appointment.uuid, it.reason) }
+          .doOnNext { cancelAppointmentWithReason.execute(it.appointment, it.reason) }
```
Comment (janhavisinghh) [inline app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt:60]: Seems like this was a part of the next commit?

**Review by janhavisinghh — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt:60]: It wasn't. This commit represents the extraction of current behaviour (cancelling an appointment) to a discrete use case. The next commit is the one in which we add an additional side effect to that use case to record a call result.

**Review by janhavisinghh — APPROVED** [review]
LGTM

**Review by janhavisinghh — APPROVED** [review]

**Pull request merged**